src/graphql/model.rs

```diff
@@ -1,38 +1,69 @@
-use crate::database::pool::PgPooledConnection;
-use crate::errors::ServiceError;
-use crate::jwt::model::{DecodedToken, HumanClaims, Token};
-use crate::user::manager::{user_manager_get_all, user_manager_get_decode, user_manager_get_jwt};
-use crate::user::model::{SlimUser, User};
+use crate::cli_args::Opt;
+use crate::database::PooledConnection;
+use crate::errors::ServiceResult;
+use crate::jwt::model::{DecodedToken, Token};
+use crate::user::model::{LoggedUser, User};
+use crate::user::service as user;
+use crate::user::service::token::ClaimsResponse;
 use juniper::Context as JuniperContext;
 use std::sync::Arc;
 
 #[derive(Clone)]
-pub struct Context {
-    pub db: Arc<PgPooledConnection>,
-    pub user: SlimUser,
+pub(crate) struct Context {
+    pub opt: Opt,
+    pub db: Arc<PooledConnection>,
+    pub user: LoggedUser,
     pub token: DecodedToken,
 }
 
 impl JuniperContext for Context {}
 
-pub struct QueryRoot;
+impl Context {
+    pub fn new(token: DecodedToken, user: LoggedUser, pool: PooledConnection, opt: Opt) -> Self {
+        Self {
+            opt,
+            token,
+            user,
+            db: Arc::new(pool),
+        }
+    }
+}
+
+pub(crate) struct QueryRoot;
 
 #[juniper::object(Context = Context)]
 impl QueryRoot {
-    pub fn users(context: &Context) -> Result<Vec<User>, ServiceError> {
-        user_manager_get_all(&context)
+    pub fn users(context: &Context, limit: Option<i32>) -> ServiceResult<Vec<User>> {
```

Comment:
Well, here it is very subjective, let me explain...

I think that looking to action parameters and validating them are a "Controller" -layer problem, and as far as my understanding of juniper / graphql works, I think this is the best place to do this.

Same goes for the `has_role` stuff.

It should then "only" calls "service" highly re-usable functions like `user::list::find_all_users`.

With this look, I think `user_manager_get_all` was doing from the parsing of the request body params ("controller" responsability, up to the DAO stuff (directly calling the diesel stuff))...